src/backend/config/deployments.py

```diff
@@ -1,140 +1,35 @@
-    ModelDeploymentName.Bedrock: Deployment(
-        id="bedrock",
-        name=ModelDeploymentName.Bedrock,
-        deployment_class=BedrockDeployment,
-        models=BedrockDeployment.list_models(),
-        is_available=BedrockDeployment.is_available(),
-        env_vars=BEDROCK_ENV_VARS,
-    ),
-}
 
+def get_installed_deployments() -> list[type[BaseDeployment]]:
+    installed_deployments = list(ALL_MODEL_DEPLOYMENTS.values())
 
-def get_available_deployments() -> dict[ModelDeploymentName, Deployment]:
-    if use_community_features:
+    if Settings().safe_lookup("feature_flags", "use_community_features"):
         try:
             from community.config.deployments import (
                 AVAILABLE_MODEL_DEPLOYMENTS as COMMUNITY_DEPLOYMENTS_SETUP,
             )
-
-            model_deployments = ALL_MODEL_DEPLOYMENTS.copy()
-            model_deployments.update(COMMUNITY_DEPLOYMENTS_SETUP)
-            return model_deployments
-        except ImportError:
+            installed_deployments = [*installed_deployments, *COMMUNITY_DEPLOYMENTS_SETUP.values()]
+        except ImportError as e:
             logger.warning(
-                event="[Deployments] No available community deployments have been configured"
+                event="[Deployments] No available community deployments have been configured", ex=e
             )
 
-    deployments = Settings().get('deployments.enabled_deployments')
-    if deployments is not None and len(deployments) > 0:
-        return {
-            key: value
-            for key, value in ALL_MODEL_DEPLOYMENTS.items()
-            if value.id in Settings().get('deployments.enabled_deployments')
-        }
-
-    return ALL_MODEL_DEPLOYMENTS
-
-
-def get_default_deployment(**kwargs) -> BaseDeployment:
-    # Fallback to the first available deployment
-    fallback = None
-    for deployment in AVAILABLE_MODEL_DEPLOYMENTS.values():
-        if deployment.is_available:
-            fallback = deployment.deployment_class(**kwargs)
-            break
-
-    default = Settings().get('deployments.default_deployment')
-    if default:
-        return next(
-            (
-                v.deployment_class(**kwargs)
-                for k, v in AVAILABLE_MODEL_DEPLOYMENTS.items()
-                if v.id == default
-            ),
-            fallback,
-        )
-    else:
-        return fallback
-
-
-def find_config_by_deployment_id(deployment_id: str) -> Deployment:
-    for deployment in AVAILABLE_MODEL_DEPLOYMENTS.values():
-        if deployment.id == deployment_id:
-            return deployment
-    return None
-
-
-def find_config_by_deployment_name(deployment_name: str) -> Deployment:
-    for deployment in AVAILABLE_MODEL_DEPLOYMENTS.values():
-        if deployment.name == deployment_name:
-            return deployment
-    return None
+    enabled_deployment_ids = Settings().safe_lookup("deployments", "enabled_deployments")
+    if enabled_deployment_ids and len(enabled_deployment_ids) > 0:
```

Comment:
I realized the previous condition had the `> 0` check, but it's not needed.

We can use `if enabled_deployment_ids` and the true case will automatically assume the list is non empty